ui/components/mission/MissionPayRedeem.tsx

```diff
@@ -466,26 +496,70 @@ export default function MissionPayRedeem({
     }
 
     try {
-      const transaction = prepareContractCall({
-        contract: primaryTerminalContract,
-        method: 'pay' as string,
-        params: [
-          mission?.projectId,
-          JB_NATIVE_TOKEN_ADDRESS,
-          inputValue * 1e18,
-          address,
-          output * 1e18,
-          message,
-          '0x00',
-        ],
-        value: BigInt(inputValue * 1e18),
-        gas: BigInt(500000),
-      })
+      if (chainSlug !== defaultChainSlug) {
+        const quoteCrossChainPay: any = await readContract({
+          contract: crossChainPayContract,
+          method: 'quoteCrossChainPay' as string,
+          params: [
+            LAYERZERO_SOURCE_CHAIN_TO_DESTINATION_EID[chainSlug].toString(),
+            inputValue * 1e18,
+            mission?.projectId,
+            address || ZERO_ADDRESS,
+            output * 1e18,
+            message,
+            '0x00',
+          ],
+        })
+        const transaction = prepareContractCall({
+          contract: crossChainPayContract,
+          method: 'crossChainPay' as string,
+          params: [
+            LAYERZERO_SOURCE_CHAIN_TO_DESTINATION_EID[chainSlug].toString(),
+            mission?.projectId,
+            inputValue * 1e18,
+            address || ZERO_ADDRESS,
+            output * 0, // Don't put in mininum output for cross-chain pay to account for slippage
+            message,
+            '0x00',
+          ],
+          value: BigInt(quoteCrossChainPay),
+        })
 
-      const receipt = await sendAndConfirmTransaction({
-        transaction,
-        account,
-      })
+        const originReceipt: any = await sendAndConfirmTransaction({
+          transaction,
+          account,
+        })
+        const destinationMessage = await waitForMessageReceived(
+          isTestnet ? 19999 : 1, // 19999 resolves to testnet, 1 to mainnet, see https://cdn.jsdelivr.net/npm/@layerzerolabs/scan-client@0.0.8/dist/client.mjs
+          originReceipt.transactionHash
+        )
+        const receipt = await waitForReceipt({
+          client: client,
+          chain: DEFAULT_CHAIN_V5,
+          transactionHash: destinationMessage.dstTxHash as `0x${string}`,
+        })
+      } else {
+        const transaction = prepareContractCall({
+          contract: primaryTerminalContract,
+          method: 'pay' as string,
+          params: [
+            mission?.projectId,
+            JB_NATIVE_TOKEN_ADDRESS,
+            inputValue * 1e18,
+            address,
+            output * 1e18,
+            message,
+            '0x00',
+          ],
+          value: BigInt(inputValue * 1e18),
+          gas: BigInt(500000),
+        })
+
+        const receipt = await sendAndConfirmTransaction({
+          transaction,
+          account,
+        })
+      }
 
       toast.success('Mission token purchased', {
```

Comment:
Not your issue - but I just noticed we're missing a period here, could you add one in. ty!

Reply:
colin beat me to it 😂